### PR TITLE
updating transit script to move root.checkpoint file from <bootstrap>…

### PR DESCRIPTION
**Pull request by vishalchangrani**
…/public-root-information to <bootstrap>/execution-state after download

Currently, we ask execution node operators to move the `root.checkpoint` to the right location after they run the `transit pull` command (https://docs.onflow.org/concepts/node-operation/quickstart/ - "_NOTE if you are running an execution node, you must also move the execution state from the public-genesis-information directory, to the execution-state directory with-in the bootstrap folder: mv ./public-root-information/root.checkpoint ./execution-state/ **This will eventually be taken care of by the transit script but currently requires manual input**._")

This change is to update the transit script to do the move instead.

**Output**:

>  go run  ./cmd/bootstrap/transit  -pull  -t mainnet-1-execution -d /tmp/bootstrap -role execution
> 2020/11/02 12:58:43 Running pull
> 2020/11/02 12:58:43 Downloading /tmp/bootstrap/public-root-information/node-infos.pub.json
> 2020/11/02 12:58:43 Downloading /tmp/bootstrap/public-root-information/root-block-seal.json
> 2020/11/02 12:58:43 Downloading /tmp/bootstrap/public-root-information/root-block.json
> 2020/11/02 12:58:43 Downloading /tmp/bootstrap/public-root-information/root-execution-result.json
> 2020/11/02 12:58:43 Downloading /tmp/bootstrap/public-root-information/root-qc.json
> 2020/11/02 12:58:43 Downloading /tmp/bootstrap/public-root-information/root.checkpoint
> 2020/11/02 12:59:41 **Moving /tmp/bootstrap/public-root-information/root.checkpoint to /tmp/bootstrap/execution-state/root.checkpoint** 
> 2020/11/02 12:59:41 MD5 of the root block is: 84591ad9b4fe4f484667a8e9bfd1efb2

### CODE REVIEW

File: cmd/bootstrap/transit/main.go [hunk 1]
```diff
@@ -458,3 +476,75 @@ func fileExists(filename string) bool {
...
+	}
+	defer destination.Close()
+
+	// open the source file
+	source, err := os.Open(src)
+	if err != nil {
+		return fmt.Errorf("failed to open file %s: %w", src, err)
+	}
+
+	// copy the source file to the destination file
```
Comment (zhangchiqing) [inline cmd/bootstrap/transit/main.go:508]: Why not using [Renaming](https://golang.org/pkg/os/#Rename), would be much faster

Comment (vishalchangrani) [reply cmd/bootstrap/transit/main.go:508]: So I came across this thread (https://stackoverflow.com/questions/50740902/move-a-file-to-a-different-drive-with-go) which talks about problems with `renaming` files if the destination is a directory on a different disk. Hence, to take the most safest approach I did a `copy, remove` as suggested over there.

**Comment by vishalchangrani** [comment]
@danuio @zhangchiqing @jkan2 @m4ksio when u have time, can you review this change? Thanks!

File: cmd/bootstrap/transit/main.go (hunk 1 above)
Comment (zhangchiqing) [reply cmd/bootstrap/transit/main.go:532]: Could we try renaming first and if it hits the error "Different disk" error, then do real copy?

I think in most cases, the root.checkpoint file will be downloaded to the same disk, no?

But this suggestion is optional.

**Review by zhangchiqing — APPROVED** [review]
Looks good. Nice improvement 👍

File: cmd/bootstrap/transit/main.go (hunk 1 above)
Comment (vishalchangrani) [reply cmd/bootstrap/transit/main.go:541]: updated

**Review by zhangchiqing — APPROVED** [review]

**Review by danuio — APPROVED** [review]

**Pull request merged**